expo/expo
# [build-properties][autolinking] add extraPods and extraMavenRepos

# Why

some third-party libraries require developers to add extra cocoapods or maven dependencies. this pr adds this customization to expo-build-properties. unlike other properties which are processed by the config-plugins, the dependencies are added by expo-modules-autolinking.
close ENG-8503

# How

- [build-properties] add `android.extraMavenRepos` and `ios.extraPods` configs and validations.
- [autolinking] read the expo-build-properties from config and add the dependencies through autolinking
- [docs] `et generate-docs-api-data -p expo-build-properties`

# Test Plan

- ci passed
- integration test on bare-expo
  1. add the following properties to **apps/bare-expo/app.json**
  ```json
    "plugins": [
      [
        "expo-build-properties",
        {
          "android": {
            "extraMavenRepos": ["https://customers.pspdfkit.com/maven/"]
          },
          "ios": {
            "extraPods": [
              {
                "name": "Protobuf"
              }
            ]
          }
        }
      ]
    ]
  ```
  2. add `implementation 'com.pspdfkit:libraries-java:1.4.1'` to **apps/bare-expo/android/app/build.gradle**
  2. Run `npx pod-install` and verify whether Protobuf is added to Podfile.lock
  3. Run `./gradlew :app:assembleDebug` and see if the pspdfkit artifact is resolved

# Checklist
